new/ohai-config.md

```diff
@@ -0,0 +1,61 @@
+---
+RFC: unassigned
+Author: Bryan McLellan <btm@loftninjas.org>
+Status: Draft
+Type: Standards Track
+---
+
+# Ohai User Configuration
+
+Provides a standard for passing configuration settings to Ohai to configure its behavior as well as that of plugins.
+
+## Motivation
+
+    As a Chef user,
+    I want to easily configure Ohai,
+    so that it behaves optimally on my infrastructure.
+
+    As an Ohai user,
+    I want Ohai to load configuration settings from my client.rb,
+    so that it behaves the same as it does during a chef-client run.
+
+## Specification
+
+### Configuration File
+
+Previously Ohai has not had a configuration file. A small number of Ohai configuration settings do exist that can be specified in the chef-client ```client.rb``` file, which are then passed to Ohai during a chef-client run. For example, additional locations on the file system can be searched for plugins by adding paths to the ```Ohai::Config[:plugin_path]``` array. This file is only loaded by Chef and does not affect Ohai when being run on the command line nor any other programs loading Ohai as a library.
+
+When run from the command line, Ohai should load the ```client.rb``` file from the appropriate platform specific path unless an alternate configuration file is provided as a command line argument. To preserve compatibility with the existing parsing of the ```client.rb```, the Ohai configuration file syntax will be variables set on the ```Ohai::Config``` class,, and barewords will be ignored using method_missing.
+
+For example:
+
+```
+Ohai::Config[:plugin_path] << "/usr/local/lib/ohai/plugins"
+Ohai::Config[:plugin][:hostname][:fqdn_using] = [ :hostname, :nis, :dns ]
+
+# This is ignored:
+node_name "darius"
+```
+
+Using the same ```client.rb``` file used by the chef-client also allows both tools to be configured simultaneously by populating this file during bootstrap with custom local values (when that is supported in the future).
+
+### Namespacing
+
+To reduce the risk of built-in and custom plugins using the same configuration setting for conflicting purposes, plugins must prefix their configuration settings with ```[:plugin]``` and the name of the plugin consuming the setting. 
```

Comment:
Makes more sense to me to use the plugin name.  A plugin might not provide a complete hash, but might fill in values inside of an array contained in a hash so it would look more `[:foo][:bar][*][:thing]`.  Just an example to show that basing the config name on the emitted JSON could get really silly.  Also the platform plugin fills in `platform` and `platform_family` so which one should we use in the config?  Do we have to support both?  What if there's 10 keys at the same level?  etc.